test/fromReader.spec.js

```diff
@@ -48,9 +48,8 @@ describe('pull-length-prefixed decodeFromReader', () => {
 
     // decode from reader
     lp.decodeFromReader(reader, function (err, output) {
-      expect(err).to.be.instanceof(Error)
-      expect(output).to.equal(undefined)
-      done()
+      expect(output).to.eql(Buffer.alloc(0))
+      done(err)
```

Comment:
I don't understand why this should work - I wouldn't have thought it possible to read from the reader since no data is being pulled from the source.

The stream should end immediately, so using `decodeFromReader` should either error or produce `null` output?

In pull stream terms, I'd expect this callback to be called with `err === true` i.e. not error, end.

Reply:
You're right, there shouldn't be any output. The callback is going to get an actual error though and not `err === true`, because it's trying to read a specific number of bytes (1) from reader, which will throw an error because it has none.